File: Algorithms/1. Sorting/bubbleSort.cpp
// how to run: c++ -std=c++11 file.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int array[8] = {1, 3, 8, 2, 9, 2, 5, 6};
  int size = 8;

  cout << "Bubble Sort O(n^2) Algorithm:\n";

  // output the array before the sort
  cout << "Array before the sort: ";
  for (int i = 0; i < size; i++)
    cout << array[i] << " ";
  cout << endl;

  // bubble sort algorithm: big-O: n^2
  for (int i = 0; i < size; i++)
    for (int j = 0; j < size-1; j++)
      if (array[j] > array[j+1])
        swap(array[j], array[j+1]);

  // output the array after the sort
  cout << "Array after the sort: ";
  for (int i = 0; i < size; i++)
    cout << array[i] << " ";
  cout << endl;
}

File: Algorithms/1. Sorting and Selection/bubble-sort/bubbleSort.py
print("\nBubble Sort O(n^2) Algorithm\n")

# create a array for testing
array = [5, 4, 3, 2, 1]

# output the array before sorting
print("Array before sorting: ")
print(array)

# perform bubble sort
length = len(array)
for i in range(length):
    for j in range(length-1):
        if array[j] > array[j+1]:
            array[j], array[j+1] = array[j+1], array[j]

# output sorted array
print("Array after sorting: ")
print(array)
